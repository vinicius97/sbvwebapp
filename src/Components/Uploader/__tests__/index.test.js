import React from 'react'
import Adapter from 'enzyme-adapter-react-16'
import { configure, shallow } from 'enzyme'
import { Uploader } from '../'

configure({adapter: new Adapter()})

describe('<Uploader />', () => {

  const wrapper = shallow(<Uploader />)

  it('Comparar snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('Validar estrutura com form oculto', () => {
    expect(wrapper.find('.uploader--form')).toHaveLength(0)
    expect(wrapper.find('.uploader--form--input__text')).toHaveLength(0)
    expect(wrapper.find('.uploader--form--input--file')).toHaveLength(0)
    expect(wrapper.find('.uploader--form__button')).toHaveLength(1)
  })

  it('Validar estrutura com form exibido', () => {
    wrapper.setState({showForm: true})
    expect(wrapper.find('.uploader--form')).toHaveLength(1)
    expect(wrapper.find('.uploader--form--input__text')).toHaveLength(1)
    expect(wrapper.find('.uploader--form--input--file')).toHaveLength(1)
    expect(wrapper.find('.uploader--form__button')).toHaveLength(2)
  })

})