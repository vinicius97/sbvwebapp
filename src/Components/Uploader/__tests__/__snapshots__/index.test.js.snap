// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Uploader /> Comparar snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Uploader
    upload={
        Object {
            "status": "",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <button
          className="uploader--form__button"
          onClick={[Function]}
          type="submit"
>
          Enviar vídeo
</button>,
        <div
          className="uploader--form--mask__hide"
          onClick={[Function]}
/>,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Enviar vídeo",
          "className": "uploader--form__button",
          "onClick": [Function],
          "type": "submit",
        },
        "ref": null,
        "rendered": "Enviar vídeo",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "uploader--form--mask__hide",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      false,
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <button
            className="uploader--form__button"
            onClick={[Function]}
            type="submit"
>
            Enviar vídeo
</button>,
          <div
            className="uploader--form--mask__hide"
            onClick={[Function]}
/>,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Enviar vídeo",
            "className": "uploader--form__button",
            "onClick": [Function],
            "type": "submit",
          },
          "ref": null,
          "rendered": "Enviar vídeo",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "uploader--form--mask__hide",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        false,
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
